libsolidity/formal/Predicate.cpp

```diff
@@ -446,3 +460,26 @@ bool Predicate::fillArray(smtutil::Expression const& _expr, vector<string>& _arr
 
 	solAssert(false, "");
 }
+
+map<string, optional<string>> Predicate::readTxVars(smtutil::Expression const& _tx) const
+{
+	map<string, TypePointer> const txVars{
```

Comment:
Couldn't this be `static`?

Reply:
At first it was, but it breaks because of `TypeProvider`'s lifetime, when running `isoltest`, for example